### PR TITLE
Improve compressed files

### PR DESCRIPTION
**Pull request by lievenhey**

### CODE REVIEW

**Comment by lievenhey** [comment]
This is a fix for the currently broken compressed files feature. The current implementation doesn't work, because I passed the compressed file to `perfparser`. This patch series fixes that and adds support for archives files so that hotspot can now open `zip` and `tar.gz` files.

**Comment by GitMensch** [comment]
Does this overlap with #633?

File: src/parsers/perf/perfparser.cpp [hunk 1]
```diff
@@ -1466,10 +1466,7 @@ PerfParser::PerfParser(QObject* parent)
         m_stopRequested = false;
     });
 
-    auto parsingStopped = [this] {
-        m_isParsing = false;
-        m_decompressed.reset();
-    };
+    auto parsingStopped = [this] { m_isParsing = false; };
```
Comment (milianw) [inline src/parsers/perf/perfparser.cpp:1474]: can we get a test for this patch please?

File: src/parsers/perf/perfparser.cpp [hunk 2]
```diff
@@ -1991,19 +1996,110 @@ void PerfParser::exportResults(const QUrl& url)
...
+    }
+
+    auto dir = archive->directory();
+    auto entries = dir->entries();
+
+    if (entries.size() == 1) {
+        return extractFile(dir, entries[0]);
+    }
+
+    for (const auto& file : {QStringLiteral("perf.data"), QStringLiteral("perf.data.perfparser")}) {
```
Comment (milianw) [inline src/parsers/perf/perfparser.cpp:2052]: reuse utility function you are adding in the other patch

File: tests/integrationtests/tst_perfparser.cpp [hunk 3]
```diff
@@ -825,6 +825,68 @@ private slots:
 
         QCOMPARE(decompressed.readAll(), QByteArrayLiteral("Hello World\n"));
     }
+
+    void testArchive_data()
```
Comment (milianw) [inline tests/integrationtests/tst_perfparser.cpp:829]: squash this into the previous patch

**Review by milianw — CHANGES_REQUESTED** [review]

File: tests/integrationtests/tst_perfparser.cpp [hunk 4]
```diff
@@ -825,6 +825,68 @@ private slots:
...
+    {
+        QTest::addColumn<QByteArray>("content");
+        QTest::addColumn<QString>("filename");
+
+        QTest::addRow("uncompressed tar") << qUncompress(QByteArray::fromBase64(QByteArrayLiteral(
```
Comment (milianw) [inline tests/integrationtests/tst_perfparser.cpp:834]: this is really ugly and opaque stuff that I feel uncomfortable with it, can we find another way?

File: tests/integrationtests/tst_perfparser.cpp [hunk 5]
```diff
@@ -825,6 +825,68 @@ private slots:
...
+        QFETCH(QByteArray, content);
+        QFETCH(QString, filename);
+
+        QTemporaryFile compressed;
+        compressed.setFileTemplate(filename);
+        compressed.open();
+        compressed.write(content);
+        compressed.close();
+
+        qDebug() << qCompress(content).toBase64();
```
Comment (milianw) [inline tests/integrationtests/tst_perfparser.cpp:882]: remove?

**Review by milianw — APPROVED** [review]

File: src/parsers/perf/perfparser.cpp (hunk 2 above)
Comment (milianw) [reply src/parsers/perf/perfparser.cpp:2057]: fine then, let's get this in as-is then!

**Comment by lievenhey** [comment]
Ok I found a way to make it work with one function.
That patch also refactors a lot of code. Mainly the use of `#include <../utils.h>` in the model classes. This is done by moving that code to the models.

**Review by milianw — APPROVED** [review]

**Pull request merged**